text/0000-track-unique-history-location-state.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 2016/12/05
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Track unique history location states
+
+# Motivation
+
+The path alone does not provide enough information. For example, if you
+visit page A, scroll down, then click on a link to page B, then click on
+a link back to page A. Your actual browser history stack is [A, B, A].
+Each of those nodes in the history should have their own unique scroll
+position. In order to record this position we need an ID that is unique
+for each node in the history.
+
+# Detailed design
+
+Code: [PR#14011](https://github.com/emberjs/ember.js/pull/14011)
+
+We simply add a `stateCounter` so we can track uniqueness on two
```

Comment:
GuidFor of the state object would assign a hidden ID to it, then forever that object would return that hidden id. Which seems like this would work, as all we are doing is assigning our own custom ID to state